lib_eio/core/eio__core.mli

```diff
@@ -250,6 +250,36 @@ module Fiber : sig
       This is just a convenience wrapper around {!fork}.
       If [fn] raises an exception then the promise is resolved to the error, but [sw] is not failed. *)
 
+  val fork_seq : sw:Switch.t -> (('a -> unit) -> unit) -> 'a Seq.t
+  (** [fork_seq ~sw fn] creates (but does not start) a new fiber to run [fn yield].
+
+      Requesting the next item from the returned sequence resumes the fiber until it
+      calls [yield x], using [x] value as the next item in the sequence. If [fn]
```

Comment:
I like this function, but is `yield` the best name for the iterator function within the closure? It's a little confusing since it's nothing like `Fibre.yield` really.  Something like `push` or `dispense` (the Seq terminology) might be clearer as the function name in the example.

Reply:
I think `yield` is the standard term here (see https://en.wikipedia.org/wiki/Generator_(computer_programming)).

e.g. in Python:

```python
def count():
    for i in range(1, 4):
        yield i

print(list(count()))
```

It's `yield` in JavaScript too: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield

`yield ()` is a bit like `Fiber.yield ()` in that it suspends the caller. The difference is that `yield` is yielding to a particular consumer, who controls when it resumes, whereas `Fiber.yield ()` immediately joins the run queue.

Reply:
I guess I'm dissatisfied that we have an entirely different terminology ("dispenser") in the Seq module, so we're not being consistent _within_ the OCaml/eio documentation. And the use of `yield` being overloaded (both control flow surrender, and argument-driven resumption) in other languages is a perennial source of confusion, so I'm not sure that's a good guideline to follow.

I just wanted to voice my vague angst here; I don't think I have a clearer alternative apart from "dispense".